### PR TITLE
Add blt_assert_exists() macro

### PR DESCRIPTION
**Pull request by gzagaris**

### CODE REVIEW

File: docs/api/utility.rst [hunk 1]
```diff
@@ -7,6 +7,42 @@ Utility Macros
...
 
+blt_assert_exists
+^^^^^^^^^^^^^^^^^^
+
+.. code-block:: cmake
+
+    blt_assert_exists(
+      [DIRECTORY <path/to/directory/]
+      [FILE <path/to/file>]
+      [TARGET <cmake-target] )
```
Comment (white238) [inline docs/api/utility.rst]: Missing > on DIRECTORY and TARGET

Comment (gzagaris) [reply docs/api/utility.rst]: Fixed. Thanks!

**Comment by gzagaris** [comment]
> Awesome! Thanks for going the extra step and doing documentation as well.

Of course! Thanks!

File: cmake/BLTMacros.cmake [hunk 2]
```diff
@@ -5,6 +5,38 @@
...
+##                    [TARGET <target>]
+##                    [FILE <path-to-file>] )
+##
+## Throws a FATAL_ERROR message if the specified directory, file, or target does
+## not exist.
+##------------------------------------------------------------------------------
+macro(blt_assert_exists)
+
+    set(options)
+    set(singleValueArgs DIRECTORY TARGET FILE EXECUTABLE )
```
Comment (kennyweiss) [inline cmake/BLTMacros.cmake]: This is a nice feature @gzagaris.

Is ``EXECUTABLE`` supposed to be here? It is not described in the documentation or handled in the code.

File: docs/api/utility.rst [hunk 3]
```diff
@@ -7,6 +7,42 @@ Utility Macros
...
+   :caption: **Example**
+   :linenos:
+
+   ## check if the directory 'blt' exists in the project
+   blt_assert_exists( DIRECTORY ${PROJECT_SOURCE_DIR}/cmake/blt )
+
+   ## check if the file 'SetupBLT.cmake' file exists
+   blt_assert_exists( FILE ${PROJECT_SOURCE_DIR}/cmake/blt/SetupBLT.cmake )
+
+   ## checks can also be bundled in one call
```
Comment (kennyweiss) [inline docs/api/utility.rst]: If so, should these be ``multiValueArgs``?

File: RELEASE-NOTES.md [hunk 4]
```diff
@@ -12,6 +12,7 @@ The project release numbers follow [Semantic Versioning](http://semver.org/spec/
 ## [Version 0.3.0] - Release date 2020-01-08
 
 ### Added
+- Added blt_assert_exists() utility macro.
```
Comment (kennyweiss) [inline RELEASE-NOTES.md]: It looks like this note was added to the wrong section -- it should be under ``Unreleased``, right?

**Pull request merged**